Cython/Compiler/Optimize.py

```diff
@@ -4181,6 +4133,73 @@ def _inject_bint_default_argument(self, node, args, arg_index, default_value):
             args[arg_index] = args[arg_index].coerce_to_boolean(self.current_env())
 
 
+def optimise_numeric_binop(operator, node, ret_type, arg0, arg1):
+    """
+    Optimise math operators for (likely) float or small integer operations.
+    """
+    # When adding IntNode/FloatNode to something else, assume other operand is also numeric.
+    # Prefer constants on RHS as they allows better size control for some operators.
+    num_nodes = (ExprNodes.IntNode, ExprNodes.FloatNode)
+    if isinstance(arg1, num_nodes):
+        if arg0.type is not PyrexTypes.py_object_type:
+            return None
+        numval = arg1
+        arg_order = 'ObjC'
+    elif isinstance(arg0, num_nodes):
+        if arg1.type is not PyrexTypes.py_object_type:
+            return None
+        numval = arg0
+        arg_order = 'CObj'
+    else:
+        return None
+
+    if not numval.has_constant_result():
+        return None
+
+    is_float = isinstance(numval, ExprNodes.FloatNode)
```

Comment:
This seems simpler (yes, I'm aware that this code was just moved around):
```suggestion
    is_float = numval.type.is_float
```